torch/csrc/jit/tensorexpr/loopnest.cpp

```diff
@@ -1662,7 +1668,11 @@ void LoopNest::rfactor(
       init_stmt = ol->cloneWithNewBody(init_stmt);
     }
 
-    parent_block->insert_stmt_before(init_stmt, new_root_for);
+    if (output_contains_target) {
+      parent_block->insert_stmt_before(init_stmt, new_root_for);
+    } else {
+      new_root_for->body()->prepend_stmt(init_stmt);
```

Comment:
It is possible for `root_for` (and thus for `new_root_for`) to contain other statement than the reduction store, right? If that's true, then we should not be inserting init in the beginning of the root-for body.

Reply:
Potentially yes, I thought about this at the time but I couldn't create an example of it with our current transformations. I might just need to try harder, or we may need something like the fuse transform to produce it.